lib/skate/detours/detours.ex

```diff
@@ -237,28 +243,49 @@ defmodule Skate.Detours.Detours do
   end
 
   @spec send_notification(
-          detour_db_result :: {:ok, Skate.Detours.Db.Detour.t()},
-          previous_status :: detour_type(),
-          user_id :: Skate.Settings.Db.User.t()
+          new_record :: Skate.Detours.Db.Detour.t() | nil,
+          previous_record :: Skate.Detours.Db.Detour.t() | nil,
+          user_id :: Skate.Settings.Db.User.id() | nil
         ) :: :ok | nil
-  @spec send_notification(
-          detour_db_result :: any(),
-          previous_status :: detour_type(),
-          user_id :: Skate.Settings.Db.User.t()
-        ) :: nil
+  @spec send_notification(%{
+          next_detour: Skate.Detours.Db.Detour.t() | nil,
+          next: detour_type() | nil,
+          previous: detour_type() | nil
+        }) :: :ok | nil
   defp send_notification(
-         {:ok, %Skate.Detours.Db.Detour{} = detour},
-         previous_status,
+         %Detour{} = new_record,
+         %Detour{} = previous_record,
          user_id
-       )
-       when previous_status != :active do
-    if categorize_detour(detour, user_id) == :active do
-      Notifications.NotificationServer.detour_activated(detour)
-    end
+       ) do
+    send_notification(%{
+      next_detour: new_record,
+      previous: categorize_detour(previous_record, user_id),
+      next: categorize_detour(new_record, user_id)
```

Comment:
non-blocking: What do you think of making `previous` / `next` into `previous_status` / `next_status` for unambiguity?